### PR TITLE
fix: db update bucket traffic by transaction

### PR DESCRIPTION
**Pull request by flywukong**
### Description

update db update bucket traffic DB table by transaction

### Rationale

avoid db error when download objects using multiple threads

### Example

NA

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

**Review by annielz — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

File: store/sqldb/traffic.go [hunk 1]
```diff
@@ -68,20 +68,11 @@ func (s *SpDBImpl) CheckQuotaAndAddReadRecord(record *corespdb.ReadRecord, quota
...
-		result := s.db.Model(&BucketTrafficTable{}).
-			Where("bucket_id = ?", bucketTraffic.BucketID).
-			Updates(BucketTrafficTable{
-				ChargedQuotaSize: quota.ChargedQuotaSize,
-				ModifiedTime:     time.Now(),
-			})
-		if result.Error != nil {
-			err = fmt.Errorf("failed to update bucket traffic table: %s", result.Error)
+		if err = s.updateChargedQuota(quota.ChargedQuotaSize, bucketTraffic); err != nil {
+			log.Errorw("fail to commit the transaction of updating bucketTraffic table, ", "error", err)
```
Comment (sysvm) [inline store/sqldb/traffic.go:72]: fix `fail` as `failed`

File: store/sqldb/traffic.go [hunk 2]
```diff
@@ -107,21 +98,11 @@ func (s *SpDBImpl) CheckQuotaAndAddReadRecord(record *corespdb.ReadRecord, quota
...
-		Updates(BucketTrafficTable{
-			ReadConsumedSize:      bucketTraffic.ReadConsumedSize,
-			FreeQuotaConsumedSize: bucketTraffic.FreeQuotaConsumedSize,
-			ModifiedTime:          time.Now(),
-		})
-	if result.Error != nil {
-		err = fmt.Errorf("failed to update bucket traffic table: %s", result.Error)
+	err = s.updateConsumedQuota(bucketTraffic)
+	if err != nil {
+		log.Errorw("fail to commit the transaction of updating bucketTraffic table, ", "error", err)
```
Comment (sysvm) [inline store/sqldb/traffic.go:103]: ditto

File: store/sqldb/traffic.go [hunk 3]
```diff
@@ -55,108 +56,164 @@ func (s *SpDBImpl) CheckQuotaAndAddReadRecord(record *corespdb.ReadRecord, quota
...
 func (s *SpDBImpl) InitBucketTraffic(bucketID uint64, bucketName string, quota *corespdb.BucketQuota) error {
-	insertBucketTraffic := &BucketTrafficTable{
-		BucketID:              bucketID,
-		FreeQuotaSize:         quota.FreeQuotaSize,
-		FreeQuotaConsumedSize: 0,
-		BucketName:            bucketName,
-		ReadConsumedSize:      0,
-		ChargedQuotaSize:      quota.ChargedQuotaSize,
-		ModifiedTime:          time.Now(),
+	tx := s.db.Begin()
```
Comment (j75689) [inline store/sqldb/traffic.go:186]: why not use the same code style `s.db.Transaction(func(tx *gorm.DB) error {...}`

Comment (flywukong) [reply store/sqldb/traffic.go]: updated

**Review by sysvm — CHANGES_REQUESTED** [review]
This pr should add UTs.

**Review by j75689 — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**